ragna/deploy/_ui/central_view.py

```diff
@@ -182,24 +182,75 @@ def __init__(self, api_wrapper, **params):
         )
         self.on_click_chat_info = None
 
+    def convert_metadata_filters_to_human(self, metadata_filters):
+        operator_map = {
+            "EQ": "==",
+            "NE": "!=",
+            "GT": ">",
+            "LT": "<",
+            "GTE": ">=",
+            "LTE": "<=",
+            "IN": "in",
+            "NOT_IN": "not in",
+            # 'RAW': 'raw'
+        }
+        result = []
+
+        def parse_dict(d):
+            if "AND" in d:
+                return f"({' AND '.join(parse_dict(item) for item in d['AND'])})"
+            if "OR" in d:
+                return f"({' OR '.join(parse_dict(item) for item in d['OR'])})"
+            for key, value in d.items():
+                if key in operator_map:
+                    # Assuming the value is a dictionary with a single key-value pair
+                    field, val = next(iter(value.items()))
+                    return f"{field} {operator_map[key]} {val}"
+
+        for key, value in metadata_filters.items():
+            result.append(parse_dict({key: value}))
+        return " AND ".join(result).strip("()")
+
     def on_click_chat_info_wrapper(self, event):
         if self.on_click_chat_info is None:
             return
 
-        pills = "".join(
-            [
-                f"""<div class='chat_document_pill'>{d['name']}</div>"""
-                for d in self.current_chat["metadata"]["documents"]
-            ]
-        )
+        if isinstance(self.current_chat["metadata"]["input"], list):
+            # Documents
+            title = "Uploaded Files"
+
+            pills = "".join(
+                [
+                    f"""<div class='chat_document_pill'>{d['name']}</div>"""
+                    for d in self.current_chat["metadata"]["input"]
+                ]
+            )
+
+            details = f"<div class='details'>{pills}</div><br />\n\n"
+            grid_height = len(self.current_chat["metadata"]["input"]) // 3
 
-        grid_height = len(self.current_chat["metadata"]["documents"]) // 3
+        elif isinstance(self.current_chat["metadata"]["input"], dict):
+            # Metadata filters
+            title = "Metadata Filters"
+            metadata_filters_readable = self.convert_metadata_filters_to_human(
+                self.current_chat["metadata"]["input"]
+            )
```

Comment:
Continuing form above, this could just be

```suggestion
            metadata_filters_readable = str(MetadataFilter.from_primitive(self.current_chat["metadata"]["input"]))
```

Reply:
I have accepted Philip's suggestion in commit 47a199.